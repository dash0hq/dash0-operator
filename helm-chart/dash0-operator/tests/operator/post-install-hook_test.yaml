suite: test post-install hook job
templates:
  - operator/post-install-hook.yaml
tests:
  - it: post-install hook job should match snapshot
    set:
      operator:
        dash0Export:
          enabled: true
          endpoint: https://ingress.dash0.com
          token: "very-secret-dash0-auth-token"
          apiEndpoint: https://api.dash0.com
    asserts:
      - matchSnapshot: {}

  - it: image tag should default to appVersion
    chart:
      version: 4.5.6
      appVersion: 99.100.101
    set:
      operator:
        dash0Export:
          enabled: true
          endpoint: https://ingress.dash0.com
          token: "very-secret-dash0-auth-token"
          apiEndpoint: https://api.dash0.com
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: ghcr.io/dash0hq/operator-controller:99.100.101

  - it: post-install hook job should not be deployed if no operator configuration auto resource is requested
    set:
      operator:
        dash0Export:
          enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should render tolerations and priority class
    set:
      operator:
        dash0Export:
          enabled: true
          endpoint: https://ingress.dash0.com
          token: "very-secret-dash0-auth-token"
          apiEndpoint: https://api.dash0.com
        tolerations:
          - key: key1
            operator: Equal
            value: value1
            effect: NoSchedule
          - key: key2
            operator: Exists
            effect: NoSchedule
        managerPriorityClassName: operator-manager-priority-class
    asserts:
      - equal:
          path: spec.template.spec.tolerations
          value:
            - key: key1
              operator: Equal
              value: value1
              effect: NoSchedule
            - key: key2
              operator: Exists
              effect: NoSchedule
      - equal:
          path: spec.template.spec.priorityClassName
          value: operator-manager-priority-class
