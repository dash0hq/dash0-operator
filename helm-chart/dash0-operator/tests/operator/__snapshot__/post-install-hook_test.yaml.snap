post-install hook job should match snapshot:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: post-install
        helm.sh/hook-delete-policy: hook-succeeded
      labels:
        app.kubernetes.io/component: installation-process
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: dash0-operator
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 0.0.0
        dash0.com/enable: "false"
        helm.sh/chart: dash0-operator-0.0.0
      name: dash0-operator-post-install
      namespace: NAMESPACE
    spec:
      backoffLimit: 2
      template:
        metadata:
          labels:
            app.kubernetes.io/component: readiness-check
            app.kubernetes.io/instance: post-install-hook
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: dash0-operator
            helm.sh/chart: dash0-operator-0.0.0
          name: RELEASE-NAME-post-install-job
        spec:
          affinity:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                  - matchExpressions:
                      - key: dash0.com/enable
                        operator: NotIn
                        values:
                          - "false"
                      - key: kubernetes.io/os
                        operator: In
                        values:
                          - linux
          automountServiceAccountToken: true
          containers:
            - command:
                - /manager
                - --auto-operator-configuration-resource-available-check
              image: ghcr.io/dash0hq/operator-controller:0.0.0
              imagePullPolicy: null
              name: post-install-job
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 256Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 500Mi
                  memory: 128Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
          restartPolicy: OnFailure
          securityContext:
            runAsNonRoot: true
          serviceAccountName: dash0-operator-controller
